config/routes.rb

```diff
@@ -274,5 +274,7 @@ def scoped_navigation_routes(context, navigation, as_redirects: false)
     end
   end
 
-  get '*unmatched_route', to: 'public_pages#page_not_found'
+  get '*unmatched_route', to: 'public_pages#page_not_found', constraints: lambda { |req|
+    req.path.exclude? 'rails/active_storage'
```

Comment:
I'd like to see this constraint only apply in dev. either way lgtm

Reply:
```
!Rails.env.development? || req.path.exclude? 'rails/active_storage'
```

I think that would accomplish my dream, in which the constraint is always true in non-dev.

Reply:
```
if Rails.env.development?
  # ActiveStorage is considered unmatched; we only use this kind of active storage routes in dev
  req.path.exclude? 'rails/active_storage'
else
  true
end
```

might be clearer

Reply:
```
if Rails.env.development?
  # In development, we the ActiveStorage .url route is "unmatched" and handled outside the app,
  # so we must skip rendering the 404 page.
  req.path.exclude? 'rails/active_storage'
else
  true
end
```